secretstores/azure/keyvault/keyvault.go

```diff
@@ -107,15 +107,15 @@ func (k *keyvaultSecretStore) BulkGetSecret(req secretstores.BulkGetSecretReques
 
 		secretResp, err := k.vaultClient.GetSecret(context.Background(), vaultURI, secretName, "")
 		if err != nil {
-			return secretstores.GetSecretResponse{}, err
+			return secretstores.BulkGetSecretResponse{}, err
 		}
 
 		secretValue := ""
 		if secretResp.Value != nil {
 			secretValue = *secretResp.Value
 		}
 
-		resp.Data[secretName] = secretValue
+		resp.Data[secretName] = map[string]string{secretName: secretValue}
```

Comment:
The logic for `secretName` is slightly different here than from the `GetSecret` codepath.. `GetSecret` uses the name that was passed in, rather than inspecting the `ID` that was returned from the server and normalizing it to a name.

This makes sense because in the `GetSecret` case the user provided the name and in the `BulkGetSecret` the user didn't. I wonder if there are subtle normalization differences that will catch people out (mainly casing).

I think regardless you've probably done the most reasonable thing here, just making an observation in case it's interesting to drill into more.

Reply:
Yep casing might be an issue here, but other than that I think it's a good solution.